test/test_utils.py

```diff
@@ -310,6 +310,21 @@ def test_template(template: str, output: str) -> None:
             True,
             id="query_function_with_whitespace",
         ),
+        pytest.param(
+            "{{ some_filter(lookup('env', 'USER')) }}",
+            True,
+            id="nested_with_filter",
```

Comment:
Filters are used with `arg | filter(more args)`, functions with `func(arg)`, so this would be nested with a function, not with a filter:
```suggestion
            "{{ some_function(lookup('env', 'USER')) }}",
            True,
            id="nested_with_function",
```